validphys2/src/validphys/filters.py

```diff
@@ -110,7 +116,12 @@ def filter_closure_data(filter_path, data, fakepdf, fakenoise, filterseed, sep_m
 
 
 def filter_closure_data_by_experiment(
-    filter_path, experiments_data, fakepdf, fakenoise, filterseed, experiments_index, sep_mult
+    filter_path, experiments_data, fakepdf, fakenoise, filterseed, experiments_index,
+    sep_mult, ADD=False, MULT=False, CORR=False, UNCORR=False, SPECIAL=False,
+    inconsistent_datasets=[], sys_rescaling_factor_1=1, sys_rescaling_factor_2=1,
+    type1_inconsistency=False, type2_inconsistency=False, reference_fit=True
+
+  
```

Comment:
Same as with `make_level1_etc`, the data should've been processed before entering this function. Specially given how far removed the arguments are from where they are used!

My personal recommendation would be to create another function of the type.

```python
def filter_closure_data_by_experiment_with_inconsistencies(**bla):
    <preprocessing of the data>
    return filter_closure_data_by_experiment(data_preprocessed, original_data=data)
```

Note that I'm changing the input data and creating a new key for original data.
It would work just the same (maybe it would be even better) to leave the original arguments as they are and add a single argument `inconsistent_data=<data already processed>`.